File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++ ) {
            map.put(nums[i], i);
        }
        for(int i = 0; i < nums.length; i++) {
            int num2 = target - nums[i];
            int otherIndex = map.get(num2);
            if (otherIndex != i ) {
                if (map.containsKey(num2)) {
                int[] result = {i , map.get(num2)};
                return result;
            }
            }
            
            
        }
        return null;
    }
}

File: Data Structures & Algorithms/contains-duplicate-ii/submission-2.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        for i in range(len(nums)):
            c = 1 
            while c <= k and i - c >= 0: 
                if (nums[i - c] == nums[i]):
                    return True
                c+=1
        return False
